tests/test_bookmarks.py

```diff
@@ -0,0 +1,143 @@
+                second_bookmark_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # collect information specific to incremental streams from sync 1 & 2
+                    replication_key = next(iter(expected_replication_keys[stream]))
+                    first_bookmark_value_utc = self.parse_date(first_bookmark_value[replication_key])
+                    second_bookmark_value_utc = self.parse_date(second_bookmark_value[replication_key])
+                    simulated_bookmark_utc = self.parse_date(simulated_states['bookmarks'][stream][replication_key])
+
+                    # verify the syncs sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_value)
+                    self.assertIsNotNone(second_bookmark_value)
+                    
+                    self.assertIsNotNone(first_bookmark_value[replication_key])
+                    self.assertIsNotNone(second_bookmark_value[replication_key])
+
+                    # verify the 2nd bookmark is equal to 1st sync bookmark
+                    self.assertEqual(first_bookmark_value[replication_key],
+                                     second_bookmark_value[replication_key])
+
+                    for record in first_sync_messages:
+                        replication_key_value_utc = self.parse_date(record.get(replication_key))
+                        # verify 1st sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value_utc, first_bookmark_value_utc,
+                                             msg="First sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+
+                    for record in second_sync_messages:
+                        replication_key_value_utc = self.parse_date(record.get(replication_key))
+                        # verify the 2nd sync replication key value is greater or equal to the 1st sync bookmarks
+                        self.assertGreaterEqual(replication_key_value_utc, simulated_bookmark_utc,
+                                                msg="Second sync records do not respect the previous bookmark")
+                        # verify the 2nd sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value_utc, second_bookmark_value_utc,
+                                             msg="Second sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+
+                    # verify that we get less data in the 2nd sync
+                    self.assertLess(second_sync_count, first_sync_count,
+                                    msg="Second sync does not have less records, bookmark usage not verified")
+
```

Comment:
current assertion list can be found here:
https://github.com/stitchdata/tap-tester/tree/master/tap_tester/base_suite_tests
It also includes type assertion (is instance of string) and format assertion as per below:
self.assertIsInstance(bookmark_value_1, str)
self.assertIsDateFormat(bookmark_value_1, self.BOOKMARK_FORMAT)
If currently_syncing is present for this tap:
// Verify sync is not interrupted by checking currently_syncing in state for sync 1
self.assertIsNone(self.currently_syncing_1)